### PR TITLE
GRP001 - Acyclic groups

### PR DESCRIPTION
**Pull request by aothms**

### CODE REVIEW

File: features/GRP001_Acyclic_groups.feature [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Feature: Acyclic groups and systems
+
+  Scenario: Agreement on IfcGroup (and hence systems) being acyclic
+
+      Then Each IfcGroup must not be referenced by itself directly or indirectly
```
Comment (JTurbakiewicz) [inline features/GRP001_Acyclic_groups.feature:5]: Hi, 

Perhaps it's possible to 'standardize' the rule a little bit?

```
@implementer-agreement
@GRP
Feature: GRP001 - Acyclic groups and systems
The rule verifies that all the IfcGroups are acyclic. 

  Scenario: Agreement on IfcGroup (and hence systems) being acyclic

      Then Each IfcGroup must not be referenced by itself directly or indirectly
  
```

File: features/steps/thens/relations.py [hunk 2]
```diff
@@ -176,3 +180,25 @@ def step_impl(context, entity, decision, relationship, preposition, other_entity
...
+    relationship = {'IfcGroup': ('HasAssignments', 'IfcRelAssignsToGroup', 'RelatingGroup')}
+    inv, ent, attr = relationship[entity]
+    
+    errors = []
+
+    def get_memberships(inst):
+        for rel in filter(misc.is_a(ent), getattr(inst, inv, [])):
+            container = getattr(rel, attr)
+            yield container
+            yield from get_memberships(container)
```
Comment (JTurbakiewicz) [inline features/steps/thens/relations.py:195]: Possibly a 'general' function to be placed somewhere in utils?

**Comment by JTurbakiewicz** [comment]
Possibly also README.MD file in the tests would be useful to keep with the convention, but I imagine time is of the essence. 

You can ping me if you'd like me to add the READMEs to the PR / PRs.

File: features/GRP001_Acyclic_groups.feature (hunk 1 above)
Comment (aothms) [reply features/GRP001_Acyclic_groups.feature]: Thanks, what was I thinking?

**Pull request merged**